### ISSUE TITLE
DynamicMapLayer and Proxy JSON returned results in Proxy Request ?undefined

### ISSUE THREAD

**Issue by larotech** [Esri/esri-leaflet#1153]
* Browser and version:
Chrome Version 71.0.3578.98

* Version of Leaflet (`L.version`):
1.5.1

* Version of esri Leaflet (`L.esri.VERSION`):
v2.2.4 - Wed Mar 20 2019 16:03:22 GMT-0700 (Pacific Daylight Time)

* ArcGIS Version
10.31

Steps to reproduce the error:

We have two ArcGIS Server environments, one for DEV and one for PROD. 
The DEV MapService reports: Supports Dynamic Layers: false
The PROD MapService reports: Supports Dynamic Layers: true

The MapService Layer '19' is being shown using:
```
L.esri.dynamicMapLayer({
                    url: overlayUrl,
                    proxy: "/proxy.ashx",
                    opacity: 1,
                    useCors: true,
                    layers: [19],
                    transparent: true,
                    format: 'png32'
                }).addTo(map);
```

When the overlayUrl is pointing to the MapService in DEV which has "Supports Dynamic Layers: false" the JSON output contains a parameter called "href" which contains the url to the export image. 

When the overlayUrl is pointing to the MapService in PROD which has "Supports Dynamic Layers: true" the JSON output contains a parameter called "imageData" which is a base64 encode of the export image. 

The ESRI Leaflet JavaScript code at: https://github.com/Esri/esri-leaflet/blob/adedb89e965fdbdb08c80c771bf245fca90f69c9/src/Layers/DynamicMapLayer.js#L185 assumes there is an output property in the returned JSON called "href" which in the case of my MapService in PROD, its not returning an "href" but instead an "imageData" property containing a base64 image. What happens in this case is this._renderImage is passed a value of undefined which results in a request of /proxy.ashx?undefined 

I propose a check for response.href or something similar to get around this issue:
```
if (this.options.proxy && response.href) {
          response.href = this.options.proxy + '?' + response.href;
        }
        if (response.href) {
          this._renderImage(response.href, bounds);
        } else {
          this._renderImage(response.imageData, bounds, response.contentType);
        }
```
This would then allow the renderImage method to look at response.imageData as href == undefined. This issue only happens for resources that need to be passed thru a proxy.

Regards,
Chris

**Comment by jgravois** [comment]
related: https://github.com/Esri/esri-leaflet/issues/847

**Comment by patrickarlt** [comment]
@gavinr it looks like https://github.com/Esri/esri-leaflet/blob/adedb89e965fdbdb08c80c771bf245fca90f69c9/src/Layers/DynamicMapLayer.js#L185 just needs to be changed to `if (this.options.proxy && response.href) {` as @larotech suggested. Assigning this to us to handle.

@larotech your suggested fix looks good. If you want to PR it that will help move things along faster.

**Comment by jwasilgeo** [comment]
Resolved in PR #1164. Thanks @crb04c!
